symphony-bdk-legacy/symphony-api-client-java/src/test/java/it/clients/symphony/api/PresenceClientTest.java

```diff
@@ -2,141 +2,536 @@
 
 import clients.symphony.api.PresenceClient;
 import clients.symphony.api.constants.PodConstants;
+import exceptions.APIClientErrorException;
+import exceptions.ForbiddenException;
+import exceptions.ServerErrorException;
+import exceptions.SymClientException;
 import it.commons.BotTest;
 
-import java.beans.Transient;
 import java.util.List;
 import java.util.UUID;
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
-import javax.ws.rs.core.HttpHeaders;
-import javax.ws.rs.core.MediaType;
+
 import model.UserPresence;
 import model.UserPresenceCategory;
 import org.junit.Before;
 import org.junit.Test;
-import static com.github.tomakehurst.wiremock.client.WireMock.*;
-import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.assertNotNull;
+
+import static org.junit.Assert.*;
 
 public class PresenceClientTest extends BotTest {
 
   private PresenceClient presenceClient;
 
   @Before
   public void initClient() {
-    this.presenceClient = new PresenceClient(this.symBotClient);
+    presenceClient = new PresenceClient(symBotClient);
   }
 
+  // getUserPresence with arguments
   @Test
   public void getUserPresenceSuccess() {
 
     stubGet(PodConstants.GETUSERPRESENCE.replace("{uid}", Long.toString(1L)).concat("?local=true"),
-        "{ \"category\": \"AVAILABLE\", \"userId\": 1, \"timestamp\": 1533928483800 }"
+        "{ "
+            + "\"category\": \"AVAILABLE\", "
+            + "\"userId\": 1, "
+            + "\"timestamp\": 1533928483800"
+            + "}"
+    );
+
+    try {
+
+      final UserPresence userPresence = presenceClient.getUserPresence(1L, true);
+      assertNotNull(userPresence);
+
+      verifyUserPresence(userPresence, 1L, UserPresenceCategory.AVAILABLE, 1533928483800L);
+
+    } catch (SymClientException e) {
+      fail();
```

Comment:
These catch...fail() statements are not useful. Test will fail if exception is thrown.

Reply:
In the method `getUserPresence` the SymClientException is not catched but thrown, so we have to catch it when we call the method. 
From the start we have been doing it like this. When a method throws an exception, we catch it.

Is it a bad way ?

Reply:
In my opinion it's good we are testing the success if any excpetion is thrown , the test have to failed because it's not a success

Reply:
As @symphony-elias said the `try/catch` block is useless. The test method can just `throw SymClientException`. The result will be the same and we will have more details in case of error than just calling `fail()`, that would give no detail regarding what happened.

Reply:
Please apply the same for following test methods, it will also improve readability 👍